### ISSUE TITLE
maven-plugin-plugin doesn't write version if they are late bound

### ISSUE THREAD

**Issue by loosebazooka** [freefair/gradle-plugins#51]
A project that contains dependency constraints
```groovy
dependency {
  constraint {
    implementation "com.google.guava:guava:27.0.1-jre"
  }
}
```

and then defines the dependency without version
```groovy
dependency {
  implementation "com.google.guava:guava"
}
```

results in a `pom.xml` with versions written out as blank
```xml
<dependency>
  <groupId>com.google.guava</groupId>
  <artifactId>guava</artifactId>
  <type>jar</type>
  <version></version>
</dependency>
```

What is expected is `<version>27.0.1-jre</version>`

This is problematic for multimodule builds that need version consistency

**Comment by loosebazooka** [comment]
Reproduced here: https://github.com/loosebazooka/maven-plugin-version-issue

**Comment by loosebazooka** [comment]
Comparing to plugin pom/jar generated through maven it looks like only the `plugin.xml` needs to include the version. The `pom.xml` is correctly generated with a `<dependencyManagement>` block.

**Comment by larsgrefer** [comment]
This should be fixed in `3.8.1`. Thanks for finding the error.
